contracts/multisig-prover/src/types.rs

```diff
@@ -63,10 +65,13 @@ impl KeyDeserialize for BatchID {
 }
 
 impl BatchID {
-    pub fn new(message_ids: &[String]) -> BatchID {
+    pub fn new(message_ids: &[String], new_worker_set: Option<WorkerSet>) -> BatchID {
```

Comment:
I don't like seeing `new_worker_set` here. I prefer treating it as a message initiated from Axelar.

Reply:
I could just move `message_ids.push(new_worker_set.hash().to_string())` to the call site. A little weird though, because it's not actually a message id, and there's not really a way to generate a proper message id from a worker set. I'm ambivalent.